test/ctest_helpers.cmake

```diff
@@ -272,6 +272,9 @@ function(define_c_test)
         if(NOT "${C_TEST_ADDITIONAL_FILES}" STREQUAL "")
             list(APPEND additional_file_args ${C_TEST_ADDITIONAL_FILES})
         endif()
+        if(C_TEST_CXX)
```

Comment:
CTest works with C++. I was just hoping to use it as it provides a nice, efficient test runner, which is already available and widely used within our code base.

But yes, there are alternatives, such as create_test_executable plus a custom shell script, so that these tests can be run more efficiently without each being a separate task in Evergreen (which is one of the advantages of CTest). There are also third party tools such as GoogleTest, but I don't think it would make sense to add a yet another dependency just for this.

But I can see how this can be confusing—even as now our CTest tests would not be confined to test/csuite.

What do others think?

Reply:
> CTest works with C++

Aha - this was something I didn't realise about this function, i.e. that the name doesn't mean "add a test written in C" - rather, it's "add a test that CTest will run". I have no issue with C++ there. Feel free to close out this thread.